nilearn/datasets/func.py

```diff
@@ -1470,24 +1471,47 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
     https://figshare.com/articles/COBRE_preprocessed_with_NIAK_0_12_4/1160600
     """
     if url is None:
-        url = "https://ndownloader.figshare.com/articles/1160600/versions/15"
-    opts = {'uncompress': True}
+        # Here we use the file that provides URL for all others
+        url = "https://figshare.com/api/articles/1160600/15/files"
 
     dataset_name = 'cobre'
     data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
                                 verbose=verbose)
-
     fdescr = _get_dataset_descr(dataset_name)
+
+    # First, fetch the file that references all individual URLs
+    files = _fetch_files(data_dir,
+                         [("files", url + "?offset=0&limit=300", {})],
+                         verbose=verbose)[0]
+    files = json.load(open(files, 'r'))
+    # Index files by name
+    files_ = {}
+    for f in files:
+        files_[f['name']] = f
+    files = files_
+
     # Fetch the phenotypic file and load it
     csv_name = 'cobre_model_group.csv'
+    csv_file = _fetch_files(
+        data_dir, [(csv_name, files[csv_name]['downloadUrl'],
+                    {'md5': files[csv_name].get('md5', None),
+                     'move': csv_name})],
+        verbose=verbose)[0]
 
-    csv_file = _fetch_files(data_dir, [(csv_name, url, opts)], verbose=verbose)
     # Load file in filename to numpy arrays
-    names = ['subject_type', 'sz', 'age', 'sex', 'fd']
-    csv_array = np.recfromcsv(csv_file[0], names=names, skip_header=True)
-    # Get the ids of the datasets
-    ids = csv_array['subject_type']
-    max_subjects = len(ids)
+    names = ['id', 'sz', 'age', 'sex', 'fd']
+    csv_array = np.recfromcsv(csv_file, names=names, skip_header=True)
+    # Change dtype of id and condition column
+    csv_array = csv_array.astype(
+        [('id', '|U17'),
+         ('sz', 'bool'),
```

Comment:
I think it is good practice to put categorical variables (`sz` and `sex`) as integers and continuous variables (`age` and `fd`) as floats. The bool was an error. Age is not really converted, numpy loads everything as float anyway.